### PR TITLE
test: Add csv reading test for `duckdb_read_csv(na.strings = )`

### PR DESCRIPTION
**Pull request by Tmonster**
prompted by this issue 
https://github.com/duckdb/duckdb/issues/8590

### CODE REVIEW

File: tests/testthat/test_read.R [hunk 1]
```diff
@@ -108,5 +108,16 @@ test_that("duckdb_read_csv() works as expected", {
   res$Species <- as.factor(res$Species)
   expect_true(identical(res, iris))
 
+  # test better na.strings handling
+  # see https://github.com/duckdb/duckdb/issues/8590
+  tf3 <- tempfile()
+  na_strings <- data.frame(num=c(1, 2, NA), char=c('yes', 'no', NA), logi=c(TRUE, FALSE, NA))
+  write.csv(na_strings, tf3, row.names = FALSE)
+  duckdb_read_csv(con, "na_table", tf3, na.strings = "-")
+  identical(
```
Comment (krlmlr) [inline tests/testthat/test_read.R:117]:
```suggestion
  expect_identical(
```

**Review by krlmlr — COMMENTED** [review]
Thanks. It's not immediately obvious what this test tests. Would it help writing a literal CSV string to a file, or pointing to a file that is already part of the codebase?

**Comment by Tmonster** [comment]
It just tests to make sure null values are properly read from a csv file using the parameter passed when `na_strings` is set

Potentially some users write to CSV using R and read using duckdb, so it's nice to make sure NA values are respected when the `na_strings` parameter is passed.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**